src/Entity.php

```diff
@@ -236,4 +248,129 @@ protected static function getReflection()
         }
         return self::$reflections[static::class];
     }
+
+    /**
+     * Entity constructor.
+     *
+     * @param array $data
+     */
+    final public function __construct(array $data = [])
+    {
+        $this->data = $this->originalData = $data;
+    }
+
+    /**
+     * Magic setter.
+     *
+     * You can overwrite this for custom functionality but we recommend not to use the properties or setter (set*)
+     * directly when they have to update the data stored in table.
+     *
+     * @param $var
+     * @param $value
+     */
+    public function __set($var, $value)
+    {
+        $col = $this->getColumnName($var);
+
+        $setter = self::forceNamingScheme('set' . ucfirst($var), static::$namingSchemeMethods);
+        if (method_exists($this, $setter) && is_callable([$this, $setter])) {
+            $oldValue = $this->__get($var);
+            $md5OldData = md5(serialize($this->data));
+            $this->$setter($value);
+            $changed = $md5OldData !== md5(serialize($this->data));
+        } else {
+            $oldValue = $this->__get($var);
+            $changed = @$this->data[$col] !== $value;
+            $this->data[$col] = $value;
+        }
+
+        if ($changed) {
+            $this->onChange($var, $oldValue, $this->__get($var));
+        }
+    }
+
+    /**
+     * Magic getter.
+     *
+     * @param $var
+     * @return mixed|null
+     */
+    public function __get($var)
+    {
+        $getter = self::forceNamingScheme('get' . ucfirst($var), static::$namingSchemeMethods);
+        if (method_exists($this, $getter) && is_callable([$this, $getter])) {
+            return $this->$getter();
+        } else {
+            $col = static::getColumnName($var);
+            return isset($this->data[$col]) ? $this->data[$col] : null;
+        }
+    }
+
+    /**
+     * Checks if entity or $var got changed.
+     *
+     * @param string $var
+     * @return bool
+     */
+    public function isDirty($var = null)
+    {
+        if ($var) {
+            $col = static::getColumnName($var);
+            return @$this->data[$col] !== @$this->originalData[$col];
+        }
+
+        return md5(serialize($this->data)) !== md5(serialize($this->originalData));
+    }
+
+    /**
+     * Resets the entity or $var to original data.
+     *
+     * @param string $var
+     */
+    public function reset($var = null)
+    {
+        if ($var) {
+            $col = static::getColumnName($var);
+            if (isset($this->originalData[$col])) {
+                $this->data[$col] = $this->originalData[$col];
+            } else {
+                unset($this->data[$col]);
+            }
+            return;
+        }
+
+        $this->data = $this->originalData;
+    }
+
+    /**
+     * Save the entity to $entityManager.
+     *
+     * @param EntityManager $entityManager
+     */
+    public function save(EntityManager $entityManager)
```

Comment:
It is not necessary - maybe another feature in the future but has nothing to do with this feature.